spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractMessageChannel.java

```diff
@@ -432,13 +432,19 @@ public boolean send(Message<?> message, long timeout) {
 					return false;
 				}
 			}
-			if (countsEnabled) {
-				metrics = channelMetrics.beforeSend();
+			if (channelMetrics.getTimer() != null && countsEnabled) {
```

Comment:
For better performance should be swapped: `if (countsEnabled && channelMetrics.getTimer() != null)`.
OTOH doesn't this look cleaner:
```
	if (countsEnabled) {
				if (channelMetrics.getTimer() != null) {
					final Message<?> messageToSend = message;
					sent = channelMetrics.getTimer().recordCallable(() -> doSend(messageToSend, timeout));
				}
				else {
					metrics = channelMetrics.beforeSend();
					sent = doSend(message, timeout);
					channelMetrics.afterSend(metrics, sent);
					metricsProcessed = true;
				}
			}
			else {
				sent = doSend(message, timeout);
			}
```
?